### PR TITLE
I18n automation

### PR DESCRIPTION
**Pull request by KraXen72**
please don't merge yet - i want to finish extracting all the screens before it's merged

the TL;DR
- migrated i18n messages to json to work with inlang (read below)
- set up the inlang Sherlock VSCode extension alongside the `i18n-js` library
    - it allows easier extraction of hardcoded keys into `translate('whatever')` or similar calls - entries in the json get automatically created based on JSON path, for example, `payCommon.status.pending` would create `{ "payCommon": { "status": { "pending": "value from string" } } }`
    - allows AI translation to different languages, as well as easy manual translations / adjustments of AI translations.
        - best part is, the AI translations are built into Sherlock, and they're free!
    - simplifies adding new languages - just create a new langauge tag in the Sherlock settings and use `{lang}.json` for the keys
        - for now, new language has to still be added to the app manually as well as imported manually
    - soon: shows the actual value next to the translate function call - waiting for https://github.com/opral/inlang-sherlock/discussions/62
    - later, the inlang ecosystem enables us to automatically spin up a web-based ui for crowd-sourced translations (fink), which correctly attribute them to each git user
- migrated a bunch of files using hardcoded strings to `tx`, `subTx` and `translate()` function calls!

### CODE REVIEW

**Review by minibits-cash — APPROVED** [review]

**Review by minibits-cash — APPROVED** [review]

**Review by minibits-cash — APPROVED** [review]

**Review by minibits-cash — COMMENTED** [review]
Shall we set to editor tabSize 2 or 4? Started with 2 then moved to 4 but never managed to change everywhere...

**Comment by KraXen72** [comment]
i fixed the merge conflict

**Comment by KraXen72** [comment]
> Shall we set to editor tabSize 2 or 4? Started with 2 then moved to 4 but never managed to change everywhere...
@minibits-cash i think for js/ts it's much better to use tab size 2 or just the tab character, but for html/jsx it's good to use 4 spaces. 
in my opinion, the best would be, either:
1) tab characters, because you can configure their size in your editor (and it's just 1 setting, dont' have to re-format the codebase)
2) set up prettier to auto-format it so it's 2 spaces for js/ts code and 4 spaces for markup/jsx, if that's even possible.

what do you think?

**Comment by KraXen72** [comment]
ok all merge conflicts should be resolved now

**Review by minibits-cash — APPROVED** [review]

**Comment by minibits-cash** [comment]
Merged
